redis_session.c

```diff
@@ -737,8 +789,14 @@ PS_READ_FUNC(redis)
     if (resp_len < 0) {
         *val = ZSTR_EMPTY_ALLOC();
     } else {
-        *val = zend_string_init(resp, resp_len, 0);
+        if (session_uncompress_data(redis_sock, resp, resp_len, &compressed_buf, &compressed_len)) { // Try uncompress
+            *val = zend_string_init(compressed_buf, compressed_len, 0);
+            efree(compressed_buf); // Free the buffer allocated by redis_uncompress
+        } else { // Return raw data
+            *val = zend_string_init(resp, resp_len, 0);
+        }
```

Comment:
If you change `session_compress_data` to always set the out arguments and return whether you need to free the char * you can eliminate this if/else and just always use `compressed_buf` and then conditionally free it based on the return value from `session_compress_data`.